core/coin.go

```diff
@@ -0,0 +1,44 @@
+package core
+
+import (
+	"errors"
+	"strings"
+)
+
+// DefaultCurrencyDivisibility - decimals for price
+const DefaultCurrencyDivisibility uint32 = 1e8
+
+var (
+	ErrUnsupportedCurrencyCode = errors.New("unsupported currency code")
+	SupportedCurrencies        = []Currency{
```

Comment:
I'm not understanding why we are hardcoding a default list of SupportedCurrencies rather than loading the list of actual currencies from the multiwallet. 

This seems like a particularly bad way of doing it as everything else in the app is designed around not hardcoding values related to wallets. 

Same with the `DefaultCurrencyDivisibility`. The currency divisibility is available from the wallet interface. I don't know why we would hardcode it in openbazaar-go.

Reply:
Because the wallet isn't guaranteed to be running any particular currency and the server would still need to provide divisibility for the currency even if the wallet doesn't know about it. (As in the case where a server is using SPV and doesn't have divisibility for ETH available there.) I don't think we should expect the currency definition to be coupled to the wallet since the wallet will not necessarily know about all of the currencies on the network.

Reply:
Neither of that sounds correct. You use it in:

1) creating the notification in the listener. You'll never being calling that function with a currency the wallet doesn't support

2) in crypto listings. Which isn't even an appropriate place to use it because the divisibility is just defaulted to the same divisibility for every coin. It doesn't vary by coin.

Reply:
Besides we were already doing this, the only reason this was extracted was because we're doing it in two places now. The only thing that's new is that I have a constructor that creates a `Currency` and decorated it with behavior `Divisibility()` instead of passing through a constant for any string input. I could relax the error case, but modularizing this behavior is the right thing to do. And if you help me understand why my case above isn't a problem, I could make it request the data from the wallet-interface instead (if you'll let me know where to retrieve it).